### PR TITLE
WAZO-1483: fix office365 search

### PR DESCRIPTION
**Pull request by bloom1**
alembic: add office365 default values
office365: fix OpenAPI spec and add description
office365: fix unhandled exception
config_service: fix office365 default config
office365_backend: aggregate all contact numbers in a field
office365_backend: remove search with term

### CODE REVIEW

File: wazo_dird/plugins/office365_backend/plugin.py [hunk 1]
```diff
@@ -77,8 +77,11 @@ def search(self, term, args=None):
             microsoft_token = self._get_microsoft_token(**args)
         except MicrosoftTokenNotFoundException:
             return []
+        try:
+            contacts = self.office365.get_contacts(microsoft_token, self.endpoint)
+        except UnexpectedEndpointException:
```
Comment (pc-m) [inline wazo_dird/plugins/office365_backend/plugin.py]: I think something should be logged here to help debugging when there's no result

File: wazo_dird/plugins/office365_backend/plugin.py [hunk 2]
```diff
@@ -101,7 +104,11 @@ def list(self, unique_ids, args=None):
         except MicrosoftTokenNotFoundException:
             return []
 
-        contacts = self.office365.get_contacts(microsoft_token, self.endpoint)
+        try:
+            contacts = self.office365.get_contacts(microsoft_token, self.endpoint)
+        except UnexpectedEndpointException:
+            return []
```
Comment (pc-m) [inline wazo_dird/plugins/office365_backend/plugin.py]: Some logging here too

File: wazo_dird/plugins/office365_backend/plugin.py [hunk 3]
```diff
@@ -125,7 +132,11 @@ def first_match(self, term, args=None):
             )
             return None
 
-        contacts = self.office365.get_contacts(microsoft_token, self.endpoint)
+        try:
+            contacts = self.office365.get_contacts(microsoft_token, self.endpoint)
+        except UnexpectedEndpointException:
+            return None
```
Comment (pc-m) [inline wazo_dird/plugins/office365_backend/plugin.py]: logging again

File: wazo_dird/plugins/office365_backend/plugin.py (hunk 1 above)
Comment (bloom1) [reply wazo_dird/plugins/office365_backend/plugin.py]: There is already a logger.error in the get_contacts method before the exception raising. Maybe we should only include more details? All it says is `An error occured while fetching information from microsoft endpoint`

Comment (pc-m) [reply wazo_dird/plugins/office365_backend/plugin.py]: You're right I missed that. We could add the status code to the current log line and maybe log the response body in a logger.debug?

Comment (pc-m) [reply wazo_dird/plugins/office365_backend/plugin.py]: I'll merge anyway since that's not something that changed from your PR.

**Pull request merged**